calculator/src/components/calculator.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import Buttons from "./buttons";
+import { evaluate } from "mathjs";
+
+class Calculator extends React.Component {
+  state = {
+    finalVal: "",
+    holder: "",
+    currentVal: "0",
+    prevVal: "",
+    opperater: "",
+    holdVal: "",
+    toggle: false,
+    clearSymbol: false
+  };
+
+  numButtonsHandler = e => {
+    let res = e.target.value;
+
+    if (this.state.opperater !== "") {
+      console.log("here1");
+      this.setState(prevState => ({
+        prevVal: prevState.currentVal,
+
+        currentVal: prevState.holdVal + res,
+        holdVal: prevState.holdVal + res
+      }));
+    } else if (this.state.currentVal === "0") {
```

Comment:
same here - `else if(!this.state.currentVal) ` . What you are doing is not wrong though :)